src/layers/src/layer-utils.ts

```diff
@@ -140,3 +249,159 @@ export function getColumnModeRequiredColumns(
 ): string[] | undefined {
   return supportedColumnModes?.find(({key}) => key === columnMode)?.requiredColumns;
 }
+
+/**
+ * Returns geoarrow fields with ARROW:extension:name POINT metadata
+ * @param fields Any fields
+ * @returns geoarrow fields with ARROW:extension:name POINT metadata
+ */
+export function getGeoArrowPointFields(fields: Field[]): Field[] {
+  return fields.filter(field => {
+    return (
+      field.type === 'geoarrow' &&
+      field.metadata?.get('ARROW:extension:name') === EXTENSION_NAME.POINT
+    );
+  });
+}
+
+/**
+ * Builds an arrow vector compatible with ARROW:extension:name geoarrow.point.
+ * @param getPosition Position accessor.
+ * @param numElements Number of elements in the vector.
+ * @returns An arrow vector compatible with ARROW:extension:name geoarrow.point.
+ */
+export function createGeoArrowPointVector(
+  dataContainer: ArrowDataContainer,
+  getPosition: ({index: number}) => number[]
+): arrow.Vector {
+  // TODO update/resize existing vector?
+  // TODO find an easier way to create point geo columns
+  // in a correct arrow format, as this approach seems too excessive for just a simple interleaved buffer.
+
+  const numElements = dataContainer.numRows();
+  const table = dataContainer.getTable();
+
+  const numCoords = numElements > 0 ? getPosition({index: 0}).length : 2;
+  const precision = 2;
+
+  const metadata = new Map();
+  metadata.set('ARROW:extension:name', EXTENSION_NAME.POINT);
+
+  const childField = new arrow.Field('xyz', new arrow.Float(precision), false, metadata);
```

Comment:
Nit: I would create something like `geoarrow-utils.ts` for the core of this function (`interleaveArrowColumns()`?) and call that function here, this would separate the arrow logic from the data container logic, and start building a reusable library of arrow table manipulation functions.